lib/backend.ml

```diff
@@ -300,3 +320,81 @@ module Regression = struct
     let f = Field.to_filename in
     Fields.to_list ~query_responses:f ~commands:f ~logs:f
 end
+
+module Benchmark = struct
+  type t = {
+    mutable bench : Benchmark_summary.t list; (* TODO: rewrite to single, not list *)
```

Comment:
Just as a side-note: if you leave this as a list, you could maybe store the different metrics in it, i.e. something like
```
[{name = "reading phase"; value = Timings <list of timings of reading phase>};
 {name = "typing phase"; value = Timings <list of timings of typing phase>};
 {name = "num spikes"; value = Num <number spikes>};
 (...)]
```
I haven't thought about it, though, so you might have something better in mind!

Reply:
I thought that it's easier to produce whole JSON in one go. Otherwise, I would need to extract all results from file and create an object later (e.g. run a post-process in OCaml, or do it via jq when running `make bench`).